### DISCUSSION TITLE
[Feature Request] Add PSI module

### DISCUSSION THREAD

**Discussion by s3rj1k** [polybar/polybar#2891] [Ideas]
Consider adding PSI (Pressure Stall Information) module, recent Linux kernels expose this information.

https://docs.kernel.org/accounting/psi.html

**Comment by patrick96** [comment]
Since this just involves periodically extracting data from a file, this would be better suited for a shell or python script instead of creating a full-blown native module.

**Reply by s3rj1k** [reply]
Sure I can do this using script, but it would be 3 exec for each CPU, MEM and IO and I would like to avoid this to keep CPU usage low just like all built-in modules do.

**Reply by patrick96** [reply]
I don't think 3 execs every 5-10 seconds is too bad. If you want even less, you can do the sleeping in your script (and use `tail = true` in the script module):

```python
while True:
    # Produce output
    # ...
    sleep(5)
```

This way, you have one initial exec which continuously produces output.

**Reply by s3rj1k** [reply]
I don't want execs at all
